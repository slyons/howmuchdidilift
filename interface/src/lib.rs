use serde::{Serialize, Deserialize};
use strum::EnumIter;

#[derive(Debug, Deserialize, Serialize)]
pub struct LoginParams {
    pub email: String,
    pub password: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct LoginResponse {
    pub token: String,
    pub pid: String,
    pub name: String,
    pub is_verified: bool,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct CurrentResponse {
    pub pid: String,
    pub name: String,
    pub email: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct RegisterParams {
    pub email: String,
    pub password: String,
    pub name: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct VerifyParams {
    pub token: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct ForgotParams {
    pub email: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct ResetParams {
    pub token: String,
    pub password: String,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct MeasureCreate {
    pub name: String,
    pub name_plural:  String,
    pub grams: f64
}

#[derive(Debug, Deserialize, Serialize, EnumIter)]
#[serde(rename_all="lowercase")]
pub enum InputWeightType {
    Lbs,
    Kgs
}

#[derive(Debug, Deserialize, Serialize)]
pub struct RandomWeightRequest {
    pub input_amt: f64,
    pub input_type: InputWeightType
}

#[derive(Debug, Deserialize, Serialize)]
pub struct RandomWeightResponse {
    pub input_amt: f64,
    pub input_type: InputWeightType,
    pub output_weight: String
}