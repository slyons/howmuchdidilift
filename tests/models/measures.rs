use insta::{assert_debug_snapshot, with_settings};
use liftcalc::app::App;
use loco_rs::testing;
use serial_test::serial;
use interface::MeasureCreate;
use liftcalc::{
    models::measures
};

macro_rules! configure_insta {
    ($($expr:expr),*) => {
        let mut settings = insta::Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("measures");
        let _guard = settings.bind_to_scope();
    };
}

#[tokio::test]
#[serial]
async fn test_model() {
    configure_insta!();

    let boot = testing::boot_test::<App>().await.unwrap();
    //testing::seed::<App>(&boot.app_context.db).await.unwrap();
    let create_params = MeasureCreate{
        name: "burger".to_string(),
        name_plural: "burgers".to_string(),
        grams: 500.0,
    };
    let create_req = measures::ActiveModel::create(&boot.app_context.db, create_params).await;
    println!("{:?}", create_req);
    assert!(create_req.is_ok());
    let create_req = create_req.unwrap();

    let item = measures::Model::find_by_name(&boot.app_context.db, "Burger")
        .await;
    assert!(item.is_ok());
    let item = item.unwrap();
    assert_eq!(create_req.id.unwrap(), item.id);
    /*with_settings!({
            filters => {
                 let mut combined_filters = testing::CLEANUP_DATE.to_vec();
                    combined_filters.extend(vec![(r#"\"id\\":\d+"#, r#""id\":ID"#)]);
                    combined_filters
            }
        }, {
            assert_debug_snapshot!(item);
        }
    );*/
    // query your model, e.g.:
    //
    // let item = models::posts::Model::find_by_pid(
    //     &boot.app_context.db,
    //     "11111111-1111-1111-1111-111111111111",
    // )
    // .await;

    // snapshot the result:
    // assert_debug_snapshot!(item);
}


