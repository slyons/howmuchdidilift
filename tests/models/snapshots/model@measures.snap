---
source: tests/models/measures.rs
expression: item
---
Model {
    created_at: DATE,
    updated_at: DATE,
    id: 1,
    name: "burger",
    name_plural: "burgers",
    grams: 500.0,
}
